### PR TITLE
Update MSAL dependency

### LINKED ISSUE #8332: Update msal dependency
azure-identity requires msal<1.0.0. We should update this when msal 1.0.0 is available.

### PR DESCRIPTION
**Pull request by chlowell**
Relaxes our `msal` dependency to accept 1.*, and adopts the latest `msal-extensions` with similarly relaxed dependency.

Closes #8332

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/setup.py [hunk 1]
```diff
@@ -73,8 +73,8 @@
     install_requires=[
         "azure-core<2.0.0,>=1.0.0",
         "cryptography>=2.1.4",
-        "msal<1.0.0,>=0.4.1",
-        "msal-extensions~=0.1.2",
+        "msal<2.0.0,>=1.0.0",
```
Comment (lmazuel) [inline sdk/identity/azure-identity/setup.py]: If there is no breaking changes, why not make it `msal<2.0.0,>=0.4.1` ? (open question)

Comment (lmazuel) [reply sdk/identity/azure-identity/setup.py]: Complementary question: are we exposing anything from msal to customers? Even thourgh callback or something?

Comment (chlowell) [reply sdk/identity/azure-identity/setup.py]: msal 0.4.1 should work but I prefer not to declare support for preview versions now that we've all moved to GA.

Nothing of msal is exposed by design, i.e. any msal in our public API is a bug 😁

Comment (johanste) [reply sdk/identity/azure-identity/setup.py]: I'm good with setting min version to 1.0.0.

**Pull request merged**